Xamarin.Forms.Core/NavigationPage.cs

```diff
@@ -364,6 +381,13 @@ void InsertPageBefore(Page page, Page before)
 				Pushed(this, args);
 		}
 
+		Task ResolveActiveNavigation(Task task)
+		{
+			if (task.Status == TaskStatus.RanToCompletion)
```

Comment:
What happens if the [task status](https://msdn.microsoft.com/en-us/library/system.threading.tasks.taskstatus(v=vs.110).aspx) is not `RanToCompletion`?